main.cpp

```diff
@@ -32,247 +83,552 @@ Create a branch named Part2
 
 //1)
 struct Skateboard
-{//5 properties:
-    //1) Grip tape color
+{
+    
     std::string gripTapeColor = "Green";
-    //2) Hardness of Wheels
+    
     int wheelsHardness = 99;
-    //3) Brand of Trucks
+    
     std::string truckBrand = "Independent";
-    //4) Age of Bearings
+    
     int bearingAgeYears = 2;
-    //5) Type of Wood
+    
     std::string woodType = "oak";
-//3 things it can do:
-    //1) roll
-    void rollDownStreet(int speed);
-    //2) slide
-    void railSlide(bool frontSide = true);
-    //3) flip
-    void kickFlip(int degree = 360);
+
+  
+    float rollDownStreet(int speed); //returns how far you roll
+    
+    double railSlide(bool frontSide = true, double velocity = 10, double friction = 10); //returns the distance slid
+    
+    bool kickFlip(int degree = 360); //returns true if the kickflip was successfully performed.
 };
 
+float Skateboard::rollDownStreet(int speed)
+{
+    if (speed > 20)
+    {
+        std::cout << "Slow Down\n.";
+        
+    }
+    
+        return speed;
+    
+    
+
+}
+
+double Skateboard::railSlide(bool frontSide, double velocity, double friction)
+{
+    if (frontSide)
+    {
+        std::cout << "Now try it backside, dude!\n.";
+    }
+    else
+    {
+        std::cout << "Now try it frontside, dude!\n.";
+    }
+    return velocity + friction;
+}
+
+bool Skateboard::kickFlip(int degree)
+{
+    if (degree == 360)
+    {
+        std::cout << "rad!\n.";
+        return true;
+    }
+    else
+    {
+        return false;
+    }
+}
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
if the `if()` block is entered, the `return aValue;` is the last line of code to be executed.
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
`return` statements are the last thing that **ever happens** in a function
That means if your `if()` block is **NOT** entered, it will be skipped.  What is the next thing that happens?  `return anotherValue;`
that means you do **NOT** need the else block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    return anotherValue;
}
```
- [ ] revise